vscode-extension/src/extension.ts

```diff
@@ -233,15 +227,27 @@ async function createNewAIConfig(
   const fileContentBuffer = await vscode.workspace.fs.readFile(fileContentPath);
   const initialContent = fileContentBuffer.toString();
 
-  const doc = await vscode.workspace.openTextDocument({
-    content: initialContent,
-    language: mode,
-  });
+  let doc: vscode.TextDocument = null;
+  let i = 0;
+  while (!doc) {
+    const fileName = `untitled${i === 0 ? "" : "-" + i}.aiconfig.${mode}`;
+    const filePath = workspaceUri
+      ? path.join(workspaceUri.path, fileName)
+      : fileName;
+    const untitledUri = vscode.Uri.file(filePath).with({ scheme: "untitled" });
+    try {
+      doc = await vscode.workspace.openTextDocument(untitledUri);
+      break;
+    } catch (err) {
+      i++;
+    }
```

Comment:
Ryan's PR has some more checks like determining if the file already exists (yours will open an existing file if that already exists instead of new one) (or at least overwrite existing file contents from lines 246 --> 249). But that's ok, we can land this and Ryan can fix forward